### PR TITLE
Remove `prospector` support in Filebeat

### PR DESCRIPTION
**Pull request by ph**
## Add CheckRemoved6xSettings and CheckRemoved6xSettings

Add new methods to verify obsolete configuration for 7.0,
added missing test for them.

## Remove the prospector option in the configuration.

In 6.3 we have deprecated the usage of the 'prospector' options, in 7.0
its now obsolete.

This commit does the following:

- Remove any backward compatibility fixes.
- Add warning when using the prospector key in configuration.
- Remove any usage of prospector in tests.
- Remove any usage or prospector in module expectation files.
- Adjust any integration tests to not check for prospector.type.
- Remove the prospector type from the fields.
- Remove any shims created by the prospector package.

### CODE REVIEW

**Comment by ph** [comment]
There is a bit more changes in the PR that I would like to, mainly due to the trailing chars, please review it with `NO WHITESPACE`

File: filebeat/fileset/modules_test.go [hunk 1]
```diff
@@ -185,27 +185,6 @@ func TestApplyOverrides(t *testing.T) {
...
-					"close_eof": true,
-				},
-				Prospector: map[string]interface{}{
-					"close_eof": true,
-				},
-			},
-		},
 		{
 			name:    "input overrides",
 			fcfg:    FilesetConfig{},
```
Comment (ph) [inline filebeat/fileset/modules_test.go]: There is another test for `inputs`.

File: filebeat/Makefile [hunk 2]
```diff
@@ -34,7 +34,7 @@ collect-docs: python-env
 	@mkdir -p docs/modules
 	@${PYTHON_ENV}/bin/python ${ES_BEATS}/filebeat/scripts/docs_collector.py --beat ${BEAT_NAME}
 
-# Generate imports for prospectors
+# Generate inputs for prospectors
```
Comment (ruflin) [inline filebeat/Makefile]:
```suggestion
# Generate imports for inputs
```

File: libbeat/common/cfgwarn/removed.go [hunk 3]
```diff
@@ -64,3 +63,23 @@ func CheckRemoved5xSetting(cfg *common.Config, setting string) error {
 
 	return fmt.Errorf("setting '%v' has been removed", current.PathOf(name))
 }
+
+// CheckRemoved5xSettings prints a warning if the obsolete setting is used.
+func CheckRemoved5xSettings(cfg *common.Config, settings ...string) error {
```
Comment (ruflin) [inline libbeat/common/cfgwarn/removed.go]: Should we remove now in master all 5.x warnings? I don't expect people to upgrade from 5.x to 7.x directly.

Comment (ph) [reply libbeat/common/cfgwarn/removed.go]: Yes I wanted to do it in another PR

**Comment by ph** [comment]
> The files that you adjusted with the whitespaces are auto genereated. Did you modify them manually? To just generated them run

No I did a quick search/replace with my editor.. and the save just fix the whitespace, I will regenerate them.

**Comment by ph** [comment]
@ruflin I've recreated the modules expected files for Filebeat, I also removed the mention of prospector for Surricata in xpack/filebeat.

It's ready for a review.

**Comment by ph** [comment]
jenkins test this please

**Comment by ruflin** [comment]
@ph I merged this one as it conflicts with all my current PR's and I would like to continue with my PR's. Hope it's ok for you.

**Comment by ph** [comment]
🤗 @ruflin All is OK I was waiting for a Jenkins build to complete.

**Pull request merged**